src/umpire/strategy/DynamicPool.cpp

```diff
@@ -79,7 +79,12 @@ DynamicPool::~DynamicPool()
     std::size_t whole_bytes;
     std::tie(addr, is_head, whole_bytes) = rec.second;
     // Deallocate if this is a whole block
-    if (is_head && bytes == whole_bytes) m_allocator->deallocate(addr);
+    if (is_head && bytes == whole_bytes) deallocateFromStrategy(addr, bytes);
+  }
+
+  if (m_used_map.size() == 0) {
```

Comment:
Why can't this be `UMPIRE_ASSERT(m_used_map.size() == 0)` ?

Reply:
Because that would be wrong. We don't always have that behavior if someone forgets to delete a pointer. If all allocations are freed, there are then certain checks we can make, which happen in debug mode.